projects/batfish/src/main/java/org/batfish/bdp/VirtualRouter.java

```diff
@@ -217,11 +219,50 @@ public void computeFib() {
     _fib = new Fib(_mainRib);
   }
 
-  public boolean computeInterAreaSummaries() {
+  /**
+   * Decides whether the current OSPF summary route metric needs to be changed based on the given
+   * route's metric.
+   *
+   * <p>Routes from the same area or outside of areaPrefix have no effect on the summary metric.
+   *
+   * @param route The route in question, whose metric is considered
+   * @param areaPrefix The Ip prefix of the OSPF area
+   * @param currentMetric The current summary metric for the area
+   * @param areaNum Area number.
+   * @return the newly computed summary metric.
+   */
+  @Nullable
+  static Long computeUpdatedOspfSummaryMetric(
+      OspfAreaRoute route, Prefix areaPrefix, @Nullable Long currentMetric, long areaNum) {
+    Prefix contributingRoutePrefix = route.getNetwork();
+    // Only update metric for different areas and if the area prefix contains the route prefix
+    if (areaNum == route.getArea() || !areaPrefix.containsPrefix(contributingRoutePrefix)) {
+      return currentMetric;
+    }
+    long contributingRouteMetric = route.getMetric();
+    // Definitely update if we have no previous metric
+    if (currentMetric == null) {
+      return contributingRouteMetric;
+    }
+    // Otherwise just take the best between route and current metrics
+    /*
+     * NOTE: The min function is used to compute the metric according to RFC 1583.
+     * HOWEVER, Cisco claims to have switched to using max function in IOS v.12.0 and later,
+     * as described in RFC 2328.
+     * (see https://www.cisco.com/c/en/us/support/docs/ip/open-shortest-path-first-ospf/7039-1.html#t29)
+     * Not sure if special handling is necessary by the OSPF process
+     */
+    return Math.min(currentMetric, contributingRouteMetric);
+  }
```

Comment:
Seems like we should switch to the newer behavior here, or be version-aware. Keeping `min` seems like the worst of most worlds :)

Reply:
(IOS 12.0 came out in 2001?)